### ISSUE TITLE
Improving Tag and Tags annotations

### ISSUE THREAD

**Issue by navidsh** [eclipse/microprofile-open-api#47]
`@Tag` and `@Tags` annotations can be defined in one JAX-RS resources class and used in other classes. In the open source Swagger annotation definition, it is confusing how to refer to tags in different annotations.

Here is the proposal:
- Adding the ref `@Tag`:
```java
org.eclipse.microprofile.openapi.annotations.tags.Tag:
public @interface Tag {
    ...
    String ref() default "";
}
```
If `ref` field is defined in a `@Tag`, there is no need to define the `name` field.

- Adding the ref s`@Tags` for easy referencing multiple tags:
```java
org.eclipse.microprofile.openapi.annotations.tags.Tags:
public @interface Tags {
    ...
    String[] refs() default {};
}
```

**Comment by navidsh** [comment]
I am working on this issue.

**Comment by arthurdm** [comment]
Solved via merge #49
